Description: use nprocs() function instead of /proc/cpuinfo
 /proc/cpuinfo varies a lot between architectures leading to the code 
 identifying zero CPUs on some systems and hence ending up reporting a cpu
 usage of -1

Author: Peter Michael Green <plugwash@debian.org>
Bug-Raspbian: https://bugs.launchpad.net/raspbian/+bug/1128012
Bug-Debian: http://bugs.debian.org/705506

--- lxtask-0.1.4.orig/src/xfce-taskmanager-linux.c
+++ lxtask-0.1.4/src/xfce-taskmanager-linux.c
@@ -28,7 +28,7 @@
 #include <glib/gi18n.h>
 #include <glib/gprintf.h>
 #include "xfce-taskmanager-linux.h"
-
+#include <sys/sysinfo.h>
 
 #if 1
 void get_task_details(gint pid,struct task *task)
@@ -378,17 +378,7 @@ gboolean get_system_status (system_statu
 
     if(!cpu_count)
     {
-        file = fopen ("/proc/cpuinfo", "r");
-        if(!file) return FALSE;
-        while (fgets (buffer, 100, file) != NULL)
-        {
-            if(buffer[0]!='p') continue;
-            if(!strncmp(buffer,"processor",9))
-            {
-                cpu_count++;
-            }
-        }
-        fclose (file);
+        cpu_count=get_nprocs();
     }
     sys_stat->cpu_count=cpu_count;
     return TRUE;
